### PR TITLE
Gamified fix logic with is complete issue fix

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Addressing the problem where only exercise components inside the swiper had isComplete set to true has been resolved. Now, all components are updated, and the swiper starts at the first incomplete component, if any, instead of the start. Additionally, there is a refactoring of the points logic, but the underlying logic remains unchanged.
Changes

## Changes
The changes are related to issue from mobile [#199](https://github.com/Educado-App/educado-mobile/issues/199)

## Related Issues

Issue https://github.com/Educado-App/educado-mobile/issues/199

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers

I have commented out some of the tests since the functions do not exist and they can't be found in the current dev branch. Please test with a new user and the third section of the course Finance

### CODE REVIEW

**Review by sofiiagran — CHANGES_REQUESTED** [review]
It looks like a lot of the tests fail? Plus it uses like 9 hours to test..?

File: helpers/answerExercises.js
Comment (Mafusn) [inline helpers/answerExercises.js:1]: Why is this deleted?

Comment (loui981c) [reply helpers/answerExercises.js:1]: all the functionality is in one place now in the file "completing.js"

File: helpers/pointSystem.js
[the remark this answers is not in the document]
Comment (loui981c) [reply helpers/pointSystem.js:1]: all the functionality is in one place now in the file "completing.js"

**Comment by Jatewo** [comment]
Failing tests and merge conflicts. Is this PR supposed to be closed?

**Pull request closed without merging**